File: backend/domain/artifact/artifact_type.py
from __future__ import annotations

from enum import StrEnum


class ArtifactType(StrEnum):
    """
    Built-in artifact types of ALOF.
    """

    RESPONSE = "response"

    LESSON = "lesson"

    SUMMARY = "summary"

    ROADMAP = "roadmap"

    QUIZ = "quiz"

    FLASHCARD = "flashcard"


class ArtifactTypes:
    """
    Artifact type helper.
    """

    @classmethod
    def values(
        cls,
    ) -> set[str]:

        return {
            artifact_type.value
            for artifact_type in ArtifactType
        }

    @classmethod
    def exists(
        cls,
        value: str,
    ) -> bool:

        return value in cls.values()

File: frontend/lib/domain/models/artifact/artifact_type.dart
enum ArtifactType {
  response('response'),
  lesson('lesson'),
  research('research'),
  summary('summary'),
  roadmap('roadmap'),
  quiz('quiz'),
  flashcard('flashcard');

  final String value;

  const ArtifactType(this.value);

  static ArtifactType fromValue(String value) {
    return ArtifactType.values.firstWhere(
      (type) => type.value == value,
      orElse: () => ArtifactType.response,
    );
  }

  static Set<String> get valuesSet {
    return ArtifactType.values.map((type) => type.value).toSet();
  }

  static bool exists(String value) {
    return valuesSet.contains(value);
  }
}
